tests/test_fastapi.py

```diff
@@ -0,0 +1,99 @@
+import ast
+from unittest.mock import patch
+
+from fastapi import FastAPI, testclient
+
+import woodchipper
+from woodchipper.configs import DevLogToStdout
+from woodchipper.context import LoggingContext, logging_ctx
+from woodchipper.http.fastapi import WoodchipperFastAPI
+
+app = FastAPI()
+WoodchipperFastAPI(app).chipperize()
+client = testclient.TestClient(app)
+
+woodchipper.configure(
+    config=DevLogToStdout,
+    facilities={"": "INFO"},
+)
+
+
+@app.get("/")
+def hello_world():
+    with LoggingContext(testvar="testval"):
+        return logging_ctx.as_dict()
+
+
+def test_fastapi_with_woodchipper(caplog):
+    with patch("woodchipper.context.os.getenv", return_value="woodchip"):
+        response = client.get("/")
+
+    assert response.status_code == 200
+    assert response.json()["woodchip.testvar"] == "testval"
+    assert response.json()["http.method"] == "GET"
+    assert response.json()["http.header.host"] == "testserver"
+    assert response.json()["http.path"] == "http://testserver/"
+
+    # These logs won't be available until after the context has exited and the response as returned
+    fastapi_colon_request_exit_log = None
+    for log in caplog.records:
+        msg = ast.literal_eval(log.message)  # weirdly the log.message is a python dict that is a string
+        if msg["event"] == "Exiting context: fastapi:request":
+            fastapi_colon_request_exit_log = msg
+            break
+
+    assert (
+        fastapi_colon_request_exit_log is not None
+    ), "An exit message matching the flask:request pattern couldn't be found"
+    assert fastapi_colon_request_exit_log["http.response.status_code"] == 200
+    assert type(fastapi_colon_request_exit_log["http.response.content_length"]) is int
+
+
+def test_fastapi_with_woodchipper_adds_query_params():
+    query_dict = {
+        "key1": "value1",
+        "key2": ["val1", "val2", "val3"],  # confirms that multiple params are handled
+        "KEY3": "value3",  # confirms that it lowercases qparams
+        "key4": "",  # confirms that it can handle empty values
+    }
+    with patch("woodchipper.context.os.getenv", return_value="woodchip"):
```

Comment:
Why do all of these need to be patched? This is primarily useful just for the single check on line 32 only (`    assert response.json()["woodchip.testvar"] == "testval"`). It will simplify the code to remove it from the rest of them.